llvm/lib/Target/AMDGPU/AMDGPUTargetMachine.cpp

```diff
@@ -1716,6 +1716,19 @@ bool GCNTargetMachine::parseMachineFunctionInfo(
     MFI->reserveWWMRegister(ParsedReg);
   }
 
+  auto setRegisterFlags = [&](const VRegInfo &Info) {
+    for (uint8_t Flag : Info.Flags) {
+      MFI->setFlag(Info.VReg, Flag);
+    }
```

Comment:
Might as well just inline this and duplicate the code, the loop is trivial enough